airflow/models/dagbag.py

```diff
@@ -434,6 +436,10 @@ def sync_to_db(self):
         # To avoid circular import - airflow.models.dagbag -> airflow.models.dag -> airflow.models.dagbag
         from airflow.models.dag import DAG
         from airflow.models.serialized_dag import SerializedDagModel
+        self.log.debug("Calling the DAG.bulk_sync_to_db method")
         DAG.bulk_sync_to_db(self.dags.values())
-        if self.store_serialized_dags:
+        # Write Serialized DAGs to DB if DAG Serialization is turned on
+        # Even though self.store_serialized_dags is False
+        if settings.STORE_SERIALIZED_DAGS:
```

Comment:
Why do we want to do this? If we do this then wouldn't _every_ dagbag start writing to the DB?

Reply:
This is only called when dagbag.sync_to_db() is called

Reply:
And whenever `sync_to_db` is called we want to write the DAGs to the DB.

Reply:
And a note: `DagBag().sync_to_db` is only called at 2 places:

1. SchedulerJob.process_file
2. in `initdb`